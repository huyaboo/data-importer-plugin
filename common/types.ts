import { CSV_FILE_TYPE, JSON_FILE_TYPE, NDJSON_FILE_TYPE } from './constants';

export type SupportedFileTypes =
  | typeof CSV_FILE_TYPE
  | typeof JSON_FILE_TYPE
  | typeof NDJSON_FILE_TYPE;
