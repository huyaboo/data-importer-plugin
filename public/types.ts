import { DataSourceManagementPluginSetup } from 'src/plugins/data_source_management/public';
import { DataSourcePluginSetup } from 'src/plugins/data_source/public';
import { NavigationPublicPluginStart } from '../../../src/plugins/navigation/public';

export interface DataImporterPluginSetupDeps {
  dataSourceManagement?: DataSourceManagementPluginSetup;
  dataSource?: DataSourcePluginSetup;
}

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface DataImporterPluginSetup {}
// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface DataImporterPluginStart {}

export interface DataImporterPluginStartDependencies {
  navigation: NavigationPublicPluginStart;
}

export interface ImportResponse {
  message: {
    total: number;
    message: string;
  };
  success: boolean;
}
