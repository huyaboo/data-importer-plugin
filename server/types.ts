import { OpenSearchClient } from 'opensearch-dashboards/server';
import { Readable } from 'stream';

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface DataImporterPluginSetup {}
// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface DataImporterPluginStart {}

export interface IngestOptions {
  client: OpenSearchClient;
  indexName: string;
  delimiter?: string;
  dataSourceId?: string;
}

export interface IngestResponse {
  total: number;
  message: string;
}

export interface ValidationOptions {
  delimiter?: string;
}

export interface IFileParser {
  validateText: (text: string, options: ValidationOptions) => Promise<boolean>;
  ingestFile: (file: Readable, options: IngestOptions) => Promise<IngestResponse>;
  ingestText: (text: string, options: IngestOptions) => Promise<IngestResponse>;
}
