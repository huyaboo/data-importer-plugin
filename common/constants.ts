export const CSV_FILE_TYPE = 'csv';
export const JSON_FILE_TYPE = 'json';
export const NDJSON_FILE_TYPE = 'ndjson';

export const SUPPORTED_FILE_TYPES = [CSV_FILE_TYPE, JSON_FILE_TYPE, NDJSON_FILE_TYPE];
