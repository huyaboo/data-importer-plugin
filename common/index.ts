export const PLUGIN_ID = 'staticDataIngestion';
export const PLUGIN_NAME = 'staticDataIngestion';
