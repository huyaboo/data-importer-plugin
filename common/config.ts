import { schema, TypeOf } from '@osd/config-schema';
import { CSV_FILE_TYPE, JSON_FILE_TYPE, NDJSON_FILE_TYPE } from './constants';

export const configSchema = schema.object({
  enabled: schema.boolean({ defaultValue: false }),
  enabledFileTypes: schema.arrayOf(schema.string(), {
    defaultValue: [CSV_FILE_TYPE, JSON_FILE_TYPE, NDJSON_FILE_TYPE],
  }),
  maxFileSizeBytes: schema.number({
    defaultValue: 100000000,
    min: 1,
  }),
  maxTextCount: schema.number({
    defaultValue: 10000,
    min: 1,
  }),
});

export type ConfigSchema = TypeOf<typeof configSchema>;
export type PublicConfigSchema = Omit<ConfigSchema, 'enabled'>;
