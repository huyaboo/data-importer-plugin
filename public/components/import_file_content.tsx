import { EuiFilePicker } from '@elastic/eui';
import React from 'react';
import { SupportedFileTypes } from '../../common/types';

export interface ImportFileContentBodyProps {
  enabledFileTypes: SupportedFileTypes[];
  onFileUpdate: (file?: File) => void;
}

export const ImportFileContentBody = ({
  enabledFileTypes,
  onFileUpdate,
}: ImportFileContentBodyProps) => {
  const acceptedFileExtensions = enabledFileTypes.map((fileType) => `.${fileType}`).join(', ');

  const onFileChange = (files: FileList | null) => {
    if (files && files.length > 0) {
      onFileUpdate(files[0]);
    } else {
      onFileUpdate(undefined);
    }
  };

  return (
    <EuiFilePicker
      fullWidth={true}
      display={'large'}
      accept={acceptedFileExtensions}
      onChange={onFileChange}
      initialPromptText={`Select or drag and drop a ${acceptedFileExtensions} file`}
    />
  );
};
