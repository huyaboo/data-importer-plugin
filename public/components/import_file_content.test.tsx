import React from 'react';
import { render } from '@testing-library/react';
import { ImportFileContentBody } from './import_file_content';
import { SUPPORTED_FILE_TYPES_LIST } from '../../common';
import { SupportedFileTypes } from '../../common/types';

jest.mock('uuid', () => ({
  v4: jest.fn(() => 'mocked-id'),
}));

describe('ImportFileContent', () => {
  it('should render', () => {
    const { container } = render(
      <ImportFileContentBody
        enabledFileTypes={SUPPORTED_FILE_TYPES_LIST as SupportedFileTypes[]}
        onFileUpdate={jest.fn()}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
