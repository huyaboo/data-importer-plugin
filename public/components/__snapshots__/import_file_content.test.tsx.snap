// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImportFileContent should render 1`] = `
<div>
  <div
    class="euiFilePicker euiFilePicker--large euiFilePicker--fullWidth"
  >
    <div
      class="euiFilePicker__wrap"
    >
      <input
        accept=".csv, .json, .ndjson"
        aria-describedby="mocked-id-filePicker__prompt"
        class="euiFilePicker__input"
        id="mocked-id"
        type="file"
      />
      <div
        class="euiFilePicker__prompt"
        id="mocked-id-filePicker__prompt"
      >
        <svg
          aria-hidden="true"
          class="euiIcon euiIcon--large euiIcon-isLoading euiFilePicker__icon"
          focusable="false"
          height="16"
          role="img"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5.277 10.088c.02.014.04.03.057.047.582.55 1.134.812 1.666.812.586 0 1.84-.293 3.713-.88L9 6.212V2H7v4.212l-1.723 3.876Zm-.438.987L3.539 14h8.922l-1.32-2.969C9.096 11.677 7.733 12 7 12c-.74 0-1.463-.315-2.161-.925ZM6 2H5V1h6v1h-1v4l3.375 7.594A1 1 0 0 1 12.461 15H3.54a1 1 0 0 1-.914-1.406L6 6V2Z"
          />
        </svg>
        <div
          class="euiFilePicker__promptText"
        >
          Select or drag and drop a .csv, .json, .ndjson file
        </div>
      </div>
    </div>
  </div>
</div>
`;
